app/controllers/actions/AuthAction.scala

```diff
@@ -67,6 +73,22 @@ class AuthActionImpl @Inject()(val authConnector: AuthConnector, val config: Con
       case _                  => Left(Redirect(routes.UnauthorisedController.onPageLoad()))
     }
   }
+
+  private def isEoriWhitelisted(enrolments: Enrolments): Boolean = {
+    val eoriOpt = enrolments.getEnrolment("HMRC-CUS-ORG").flatMap(_.getIdentifier("EORINumber"))
+
+    eoriOpt.exists(eori => eoriWhitelist.allows(eori.value))
+  }
 }
 
 trait AuthAction extends ActionBuilder[AuthenticatedRequest, AnyContent] with ActionRefiner[Request, AuthenticatedRequest]
+
+@ProvidedBy(classOf[EoriWhitelistProvider])
+class EoriWhitelist(val values: Seq[String]) {
+  def allows(eori: String): Boolean = values.isEmpty || values.contains(eori)
```

Comment:
Is that right?  No white list means everyone is allowed?

Reply:
Yes, the same approach is in Exports service